dbms/src/Common/FailPoint.cpp

```diff
@@ -85,36 +85,42 @@ std::unordered_map<String, std::shared_ptr<FailPointChannel>> FailPointHelper::f
     M(force_remote_read_for_batch_cop)                       \
     M(force_context_path)                                    \
     M(force_slow_page_storage_snapshot_release)              \
-    M(force_change_all_blobs_to_read_only)
+    M(force_change_all_blobs_to_read_only)                   \
+    M(unblock_query_init_after_write)
+
+
+#define APPLY_FOR_FAILPOINTS_ONCE_WITH_PAUSEABLE(M) \
+    M(pause_with_alter_locks_acquired)              \
+    M(hang_in_execution)                            \
+    M(pause_before_dt_background_delta_merge)       \
+    M(pause_until_dt_background_delta_merge)        \
+    M(pause_before_apply_raft_cmd)                  \
+    M(pause_before_apply_raft_snapshot)             \
+    M(pause_until_apply_raft_snapshot)              \
+    M(pause_after_copr_streams_acquired_once)
 
+/// the usage of pause_query_init and unblock_query_init_after_write :
+/// the failpoint pause_query_init should use with the failpoint unblock_query_init_after_write, to fulfill that the select query action will be blocked before init state to wait the write action finished.
```

Comment:
Please wrap rows so that reading the comment does not need to horizontally scroll the screen.